### PR TITLE
add enumerate

### PR DESCRIPTION
**Pull request by ratmice**
Here is a quick implementation of enumerate. The tests are I guess a bit wonky, but I wanted to test that we match the behavior for some of the `std::iter::Enumerate`'s non-fused corner cases.

### CODE REVIEW

File: src/adapters/enumerate.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    pub(crate) fn new(iter: I) -> Self {
+        Enumerate { iter, count: 0 }
+    }
+}
+
+impl<I: LendingIterator> LendingIterator for Enumerate<I> {
+    type Item<'a> = (usize, I::Item<'a>) where Self: 'a;
+    fn next(&mut self) -> Option<Self::Item<'_>> {
+        let count = self.count;
+        self.iter.next().map(|item| {
```
[the remark this answers is not in the document]
Comment (Crazytieguy) [reply src/adapters/enumerate.rs:19]: I think I prefer std's version

**Comment by Crazytieguy** [comment]
Looks great! Sorry for the delay on the other PR, will try getting to it today

File: src/adapters/enumerate.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ratmice) [reply src/adapters/enumerate.rs]: Switched to that.

**Comment by ratmice** [comment]
Don't worry about delays at all, I'm just happy to help.

I went ahead and rebased out the interim change.  I'm happy to merge as-is, but will leave it for a day or so in case anybody can think of anything else?

**Comment by ratmice** [comment]
Went ahead and merged this one.
